### PR TITLE
Added unit tests for coverage

### PR DESCRIPTION
**Pull request by bosesuneha**
# Description

Added unit tests for increasing code coverage.

Fixes # (issue)
Feature # (details)

## Type of change

Please delete options that are not relevant.

-  [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Is it a breaking change which will impact consuming tool(s).

- Unit testing

# Checklist:

-  [x] My code follows the style guidelines of this project
-  [x] I have performed a self-review of my code
-  [ ] I have commented my code, particularly in hard-to-understand areas
-  [ ] I have made corresponding changes to the documentation
-  [x] My changes generate no new warnings
-  [x] I have added tests that prove my fix is effective or that my feature works
-  [x] New and existing unit tests pass locally with my changes
-  [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/test/suite/commands/runDraftTool/helper/draftCommandBuilder.test.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+         lang,
+         port,
+         appName,
+         workflow,
+         langVersion,
+         namespace,
+         imageName,
+         imageTag
+      );
+      assert.match(result, /^.*\.(yaml)$/);
```
Comment (OliverMKing) [inline src/test/suite/commands/runDraftTool/helper/draftCommandBuilder.test.ts:51]: we should check the actual contents of the create config yaml file here as well

File: src/test/suite/commands/runDraftTool/helper/runDraftHelper.test.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+} from '../../../../../commands/runDraftTool/helper/runDraftHelper';
+import {
+   getAsyncResult,
+   failed,
+   succeeded
+} from '../../../../../utils/errorable';
+
+suite('Run Draft Helper Test Suite', () => {
+   test('it returns true when draft binary exists', async () => {
+      var ensureDraftBinaryResult = await ensureDraftBinary();
```
Comment (OliverMKing) [inline src/test/suite/commands/runDraftTool/helper/runDraftHelper.test.ts:15]: You should pretty much never use `var` with modern JavaScript. https://www.freecodecamp.org/news/var-let-and-const-whats-the-difference/

Always prefer `const` > `let` > `var`

File: src/test/suite/component/download.test.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import * as assert from 'assert';
+import {toTempFile, once} from '../../../component/download';
+import {succeeded} from '../../../utils/errorable';
+import * as fs from 'fs';
+
+suite('Run Download Test Suite', () => {
+   test('it creates temp file from given source url', async () => {
+      var result = await toTempFile(
+         'https://github.com/Azure/aks-devx-tools/blob/main/README.md'
+      );
```
Comment (OliverMKing) [inline src/test/suite/component/download.test.ts:10]: We should maybe read the contents of the temp file as well and validate them

**Review by OliverMKing — CHANGES_REQUESTED** [review]
Good improvements! Left a few small comments.

File: package.json [hunk 4]
```diff
@@ -185,7 +185,8 @@
       "pretest": "npm run compile-tests && npm run compile && npm run lint",
       "lint": "eslint src --ext ts",
       "format": "prettier --write .",
-      "test": "node ./out/test/runTest.js"
+      "test": "node ./out/test/runTest.js",
+      "coverage": "c8 --check-coverage --timeout 10000 npm run test"
```
[the remark this answers is not in the document]
Comment (bosesuneha) [reply package.json:189]: Added it as a dev dependency.

File: src/test/suite/commands/runDraftTool/helper/runDraftHelper.test.ts (hunk 2 above)
Comment (bosesuneha) [reply src/test/suite/commands/runDraftTool/helper/runDraftHelper.test.ts]: thanks for pointing out! I've changed them to const.

File: src/test/suite/commands/runDraftTool/helper/draftCommandBuilder.test.ts (hunk 1 above)
Comment (bosesuneha) [reply src/test/suite/commands/runDraftTool/helper/draftCommandBuilder.test.ts:51]: added a validation for the yaml contents in the file.

File: src/test/suite/component/download.test.ts (hunk 3 above)
Comment (bosesuneha) [reply src/test/suite/component/download.test.ts]: added validation for the contents of the downloaded file.

**Review by OliverMKing — APPROVED** [review]
Looks good! Great additions!

**Review by bfoley13 — APPROVED** [review]
Nice additions of tests!

**Pull request merged**